spec/build/script/csharp_spec.rb

```diff
@@ -25,6 +25,22 @@
     end
   end
 
+  describe 'version switching' do
+    it 'throws a error with a invalid version' do
+      data[:config][:mono] = 'foo'
+      should include_sexp [:echo, 'foo is not a valid version of mono.', {:ansi=>:red}]
+    end
```

Comment:
might make sense to add a test that uses an invalid, but number-like version, e.g. 99.1000.1